docs/languages/TypeScript.md

```diff
@@ -26,6 +27,27 @@ Sometimes you don't care about classes, but rather have interfaces generated. Th
 
 Check out this [example out for a live demonstration](../../examples/typescript-interface).
 
+## Generate different `mapType`s for an `object`
+
+Typescript offers different `mapType`s which can simplify the use based on the needs. This behavior can be changed through the [`mapType` configuration](https://github.com/asyncapi/modelina/blob/master/docs/generators.md#typescript).
+
+- Use `Map` when you need a dynamic collection of key-value pairs with built-in methods for manipulation.
+- Use `Record` when you want to define an object with specific keys and their corresponding value types.
+- Use `IndexedObject` (or an interface with index signature) for a more generic approach when working with objects with dynamic keys.
+
+An example of the generated code can be seen below:
+
+```ts
+  // mapType = indexedObject
+  private _person?: { [name: string]: any };
+
+  // mapType = Map
+  private _person?: Map<string, any>;
+
+  // mapType = Record
+    private _person?: Record<string, any>;
+```
+
```

Comment:
Please use the proper values, think its all lower case 🤔 But do double check 👍

Also make sure you link to the example